src/Utils/BlockTemplateUtils.php

```diff
@@ -197,7 +197,7 @@ public static function build_template_result_from_file( $template_file, $templat
 		$template->source         = $template_file->source ? $template_file->source : 'plugin';
 		$template->slug           = $template_file->slug;
 		$template->type           = $template_type;
-		$template->title          = ! empty( $template_file->title ) ? $template_file->title : self::convert_slug_to_title( $template_file->slug );
+		$template->title          = ! empty( $template_file->title ) ? $template_file->title : self::get_block_template_title( $template_file->slug );
```

Comment:
I think that we should have only one place for editing the title and description.  I prefer to not edit this function and move all the new logic in the `add_block_templates` function.

Reply:
Not sure I understand this piece of feedback. Could you please elaborate?

Reply:
I think this looks fine to me @gigitux unless I am missing something? If I understand it correctly we are just checking for a title in `$plugin_template_types` and if one does not exist for the given `template_slug` we default to converting the slug into a human readable title anyway.

Reply:
> I think this looks fine to me @gigitux unless I am missing something? 

Before the commit [d611f6b](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/6345/commits/d611f6bee61654594fb6a67f8a2e6bdd68d440ab), we set the title and description in two different places in the codebase. My idea is to have just one place for setting everything. Now, it is fine for me :)